test/Microsoft.Azure.WebJobs.Host.UnitTests/ConfigurationUtilityTests.cs

```diff
@@ -60,5 +60,41 @@ public void GetSettingFromConfigOrEnvironment_ConfigAndEnvironment_ConfigWins()
 
             Environment.SetEnvironmentVariable("DisableSetting0", null);
         }
+
+        [Theory]
+        [InlineData("True", true)]
+        [InlineData("true", true)]
+        [InlineData("", false)]
+        [InlineData("false", false)]
+        [InlineData("test", false)]
+        public void IsDisabledMethod_ReturnsExpected(string settingValue, bool expectedResult)
+        {
+            // Arrange
+            ConfigurationUtility.Reset();
+            Environment.SetEnvironmentVariable("TestMethod.Disabled", settingValue);
```

Comment:
>Environment.SetEnvironmentVariable [](start = 12, length = 34)

Can we avoid setting an Env var? That can lead to test brittleness. 
I see we try to reset it below - at leas that should be in a finally block.

Reply:
Finally block was added.

Reply:
I think one of the benefits of using `IConfiguration` is that we should be able to supply our own `IConfigurationBuilder` for testing -- you can use an in-memory dictionary for example -- and not worry about environment vars or files, etc. Do we need to refactor ConfigurationUtility so we can plug in a custom builder via tests? @fabiocav?

Reply:
Oh -- I see we have `SetConfigurationFactory()` already. Should probably take advantage of that instead of using Environment vars for unit tests. I also see that it uses a static factory Func without a way to reset it to the default behavior which may be confusing in tests -- so maybe there's some refactoring that still needs to be done. Should it be an instance class that you create rather than static like this?